### ISSUE TITLE
does Style/PredicateName make sense for controllers

### ISSUE THREAD

**Issue by Ana06** [bbatsov/rubocop#4479]
Style/PredicateName complain about controller methods, suggesting to use names like `even?`. Do we really want to have exclamation marks in controller methods names?

--------

## Actual behavior

Style/PredicateName complain about controller methods, saying that you should replace `is_even` by `even?`.

## Expected behavior

Do not complain for controllers or not suggesting to use `?` in controller method names.

## Steps to reproduce the problem

Call `is_even` to a controller method and run rubocop :wink: 

## RuboCop version

```
0.47.1 (using Parser 2.3.3.1, running on ruby 2.4.1 x86_64-linux-gnu)
```

**Comment by Drenmi** [comment]
No. Controller actions should probably not end with an `?`. 🙂 

The general guideline is to not make special cases for different libraries in RuboCop core. Ideally, this should be handled by library specific extensions, but at the moment there is no extension for Rails.

Until there is a fleshed out extension system and/or a Rails extension, I think the best bet is to add `app/controllers/**/*` to `Exclude` for this cop in the local configuration file. 🙂

**Comment by Ana06** [comment]
@Drenmi 

> Until there is a fleshed out extension system and/or a Rails extension, I think the best bet is to add app/controllers/**/* to Exclude for this cop in the local configuration file. 

But that means that everybody using Rubocop in a Rails project should do this manually. I really don't like it.. And in this case, this should be maybe documented somewhere. :thinking: 

> The general guideline is to not make special cases for different libraries in RuboCop core. Ideally, this should be handled by library specific extensions, but at the moment there is no extension for Rails.

Is there any library specific extension? I mean other than Rails, just to see what you exactly mean.

**Comment by Drenmi** [comment]
> But that means that everybody using Rubocop in a Rails project should do this manually. I really don't like it..

I will try to clarify, to the best of my understanding. 🙂 

RuboCop is a Ruby static code analyser.

Because of Ruby's dynamic nature, accounting for different libraries' monkey patches, DSLs, etc. would confound this goal a lot. If we start digging, there are hundreds of special cases **per library**. 🙂 

There can also be countless conflicts between different libraries, or even user code, e.g. MVCs that are fine with predicate controller methods.

The option is kind of having to say that `ruby == rails`, and add library-specific considerations out of the box.

That said, there **have been** exceptions to this, when it comes to non-code configuration. 🙂 

If you look in the default config file, there are a few exclusions of `spec/**/*`. This is because we got a lot of issues opened by people using RuboCop with RSpec, and the cops didn't play nice with the DSL. The amount of inconvenience outweighed the downsides of having that configuration. (This is usually judged on a case-by-case basis.)

The best way to make your case would be to open a PR with the suggested default configuration, and have this case considered. 😊 

> Is there any library specific extension? I mean other than Rails, just to see what you exactly mean.

Have a look at [rubocop-rspec](https://github.com/backus/rubocop-rspec). It's a great example. 🙂

**Comment by Ana06** [comment]
@Drenmi 

> The best way to make your case would be to open a PR with the suggested default configuration, and have this case considered. 

Then I misunderstood you. Great! I'll open the PR. :wink: 

> Have a look at rubocop-rspec. It's a great example. 

Thanks! :wink:

**Comment by bbatsov** [comment]
I don't agree with the discussion so far - you can easily have a ton of helper boolean methods in a controller class. On the other side - a controller action like `is_something` is pretty rare and not exactly in line with REST. I've never written such a method in my entire career. Can you give me some examples when something like this is a good idea?

**Comment by mikegee** [comment]
> a controller action like `is_something` is pretty rare and not exactly in line with REST.

I just saw an `is_archived` controller action in the code I work on. I don't like it. I think it should have been an attribute of a resource.

I'm in favor of leaving the cop the way it is and dropping an inline `rubocop:disable` comment on these controller actions when needed. Better yet, refactor them away.

**Comment by deivid-rodriguez** [comment]
I agree with the latest two comments.

Also, I've never tried it so I might be wrong. But but I don't think Rails will care if you name a controller action with an ?-ended method. As long as it doesn't match a URL, I don't see a reason why it shouldn't work. I mean, `get :even?` will probably not work, but `get :is_even, to: "dice#even?"` might actually work.

In summary, I think this is out of RuboCop's scope. Even a default config.

**Comment by bbatsov** [comment]
OK, seems we can close this then.
